datatypes/src/collections/multi_polygon_collection.rs

```diff
@@ -334,8 +335,76 @@ impl ReplaceRawArrayCoords for MultiPolygonCollection {
     }
 }
 
+impl From<Vec<geo::MultiPolygon<f64>>> for MultiPolygonCollection {
+    fn from(geo_polygons: Vec<geo::MultiPolygon<f64>>) -> Self {
+        let mut features = vec![];
+        let mut polygons = vec![];
+        let mut rings = vec![];
+        let mut coordinates = vec![];
+
+        for multi_polygon in geo_polygons {
+            features.push(polygons.len() as i32);
+            for polygon in multi_polygon {
+                polygons.push(rings.len() as i32);
+                rings.push(coordinates.len() as i32);
+
+                let mut outer_coords: Vec<Coordinate2D> =
+                    polygon.exterior().points_iter().map(From::from).collect();
+
+                coordinates.append(&mut outer_coords);
+
+                for inner_ring in polygon.interiors() {
+                    rings.push(coordinates.len() as i32);
+                    let mut inner_coords: Vec<Coordinate2D> =
+                        inner_ring.points_iter().map(From::from).collect();
+
+                    coordinates.append(&mut inner_coords);
+                }
+            }
+        }
+
+        features.push(polygons.len() as i32);
+        polygons.push(rings.len() as i32);
+        rings.push(coordinates.len() as i32);
+
+        let mut builder = FeatureCollection::<MultiPolygon>::builder()
+            .batch_builder(features.len() - 1, coordinates.len() - 1);
```

Comment:
because you input the number of features, but the array with the feature offsets contains an additional entry that points to the end of the rings array.